cms/djangoapps/contentstore/tests/test_transcripts_utils.py

```diff
@@ -636,3 +637,64 @@ def test_unicode(self):
         self.assertEqual(name, u'subs_˙∆©ƒƒƒ.srt.sjson')
         name = transcripts_utils.subs_filename(u"˙∆©ƒƒƒ", 'uk')
         self.assertEqual(name, u'uk_subs_˙∆©ƒƒƒ.srt.sjson')
+
+
+@ddt.ddt
+class TestGetTranscriptsData(unittest.TestCase):
+    """
+    Tests for get_transcript_data
+    """
+    @ddt.data(
+        (
+            {
+                'edx_video_id': '000-000-000',
+                'youtube_id_1_0': '12as34',
+                'html5_sources': [
+                    'www.abc.com/foo.mp4', 'www.abc.com/bar.webm', 'foo/bar/baz.m3u8'
+                ],
+            },
+            lambda **kwargs: {
+                ('000-000-000', 'en'): 'EDX_VIDEO_ID_SUBS',
+                ('12as34', 'en'): 'YOUTUBE_SUBS',
+                ('foo', 'en'): 'HTML5_SUBS'
+            }[(kwargs['video_id'], kwargs['language_code'])],
+            'EDX_VIDEO_ID_SUBS',
+        ),
+        (
+            {
+                'edx_video_id': '',
+                'youtube_id_1_0': '12as34',
+                'html5_sources': [
+                    'www.abc.com/foo.mp4', 'www.abc.com/bar.webm', 'foo/bar/baz.m3u8'
+                ],
+            },
+            lambda **kwargs: {
+                ('12as34', 'en'): 'YOUTUBE_SUBS',
+                ('foo', 'en'): 'HTML5_SUBS'
+            }[(kwargs['video_id'], kwargs['language_code'])],
+            'YOUTUBE_SUBS',
+        ),
+        (
+            {
+                'edx_video_id': '',
+                'youtube_id_1_0': '',
+                'html5_sources': [
+                    'www.abc.com/foo.mp4', 'www.abc.com/bar.webm',
+                ],
+            },
+            lambda **kwargs: {
+                ('foo', 'en'): 'HTML5_SUBS'
+            }[(kwargs['video_id'], kwargs['language_code'])],
+            'HTML5_SUBS',
+        )
+    )
```

Comment:
What about a case where no HTML5_SUBS as well ? Also, may be we should Also consider a case where no HTML5_SUBS but edx video id and youtube and those kinda use cases as well ?

Reply:
Fair enough.